### PR TITLE
Fetch cart: feature switcher

### PR DESCRIPTION
**Pull request by garmashdev**
# Description
Added new feature switcher: 'M2_FETCH_CART_VIA_API' which is disabling "bolt_cart" customer section.

Fixes: https://app.asana.com/0/1201931884901947/1203149984454972/f

#changelog new feature switcher M2_FETCH_CART_VIA_API

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Comment by garmashdev** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1660** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1660" target="_blank">Graphite</a> 
    * **PR #1667** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1667" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1667?utm_source=stack-comment).

File: Helper/FeatureSwitch/Definitions.php [hunk 1]
```diff
@@ -296,12 +296,17 @@ class Definitions
...
+
     /**
      * Skip cart discount total mismatch
      */
     const M2_SKIP_CART_DISCOUNT_TOTAL_MISMATCH = 'M2_SKIP_CART_DISCOUNT_TOTAL_MISMATCH';
 
+    /**
+     * Enable API driven cart integration
+     */
+    const M2_ENABLE_API_DRIVEN_CART_INTEGRATION = 'M2_ENABLE_API_DRIVEN_CART_INTEGRATION';
```
Comment (vitaliyreznikov) [inline Helper/FeatureSwitch/Definitions.php:308]: Let's discuss the best name with the team.
This one can be a little bit confusing.

Comment (vitaliyreznikov) [reply Helper/FeatureSwitch/Definitions.php:308]: Let's rename it to M2_FETCH_CART_VIA_API

Comment (garmashdev) [reply Helper/FeatureSwitch/Definitions.php]: Renamed

**Review by vitaliyreznikov — APPROVED** [review]

**Pull request merged**